--- venv2/lib/python3.12/site-packages/xlsx2csv.py	2025-04-05 09:48:44.039777451 +0100
+++ venv/lib/python3.12/site-packages/xlsx2csv.py	2025-03-28 06:48:24.900097405 +0000
@@ -615,6 +615,12 @@
             self.rPh = False
 
 
+def format_timedelta(td: datetime.timedelta) -> str:
+    total_seconds = int(td.total_seconds())
+    hours, remainder = divmod(total_seconds, 3600)
+    minutes, seconds = divmod(remainder, 60)
+    return f"{hours}:{minutes:02}:{seconds:02}"
+
 class Sheet:
     def __init__(self, workbook, sharedString, styles, filehandle):
         self.py3 = sys.version_info[0] == 3
@@ -829,7 +835,9 @@
             elif self.colType == "n":
                 format_type = "float"
 
-            if format_type and not format_type in self.ignore_formats:
+            if data == '#N/A':
+                self.data = data
+            elif format_type and not format_type in self.ignore_formats:
                 try:
                     if format_type == 'date':  # date/time
                         if self.workbook.date1904:
@@ -850,9 +858,9 @@
                                 .replace(":mm", ":%M").replace("m", "%m").replace("%m%m", "%m")
                             self.data = date.strftime(str(dateformat)).strip()
                     elif format_type == 'time':  # time
-                        t = int(round((float(self.data) % 1) * 24 * 60 * 60, 6))  # it should be in seconds
-                        d = datetime.time(int((t // 3600) % 24), int((t // 60) % 60), int(t % 60))
-                        self.data = d.strftime(self.timeformat)
+                        t = int(round((float(self.data)) * 24 * 60 * 60, 6))  # it should be in seconds
+                        d = datetime.timedelta(hours=int((t // 3600)), minutes=int((t // 60) % 60), seconds=int(t % 60))
+                        self.data = format_timedelta(d)
                     elif format_type == 'float' and ('E' in self.data or 'e' in self.data):
                         self.data = str(self.floatformat or '%f') % float(self.data)
                     # if cell is general, be aggressive about stripping any trailing 0s, decimal points, etc.
@@ -871,7 +879,7 @@
                         self.data = ("%f" % (float(self.data))).rstrip('0').rstrip('.')
 
                 except (ValueError, OverflowError):  # this catch must be removed, it's hiding potential problems
-                    raise XlsxValueError("Error: potential invalid date format.")
+                    raise XlsxValueError(f"Error: potential invalid {format_type} format. - {format_str}")
 
     def handleStartElement(self, name, attrs):
         has_namespace = name.find(":") > 0
@@ -890,7 +898,8 @@
                     (name == 'v' or name == 'is') or (has_namespace and (name.endswith(':v') or name.endswith(':is')))):
             self.in_cell_value = True
             self.collected_string = ""
-        elif self.in_sheet and (name == 'row' or (has_namespace and name.endswith(':row'))) and ('r' in attrs) and not ('hidden' in attrs and attrs['hidden'] == '1'):
+        elif self.in_sheet and (name == 'row' or (has_namespace and name.endswith(':row'))) and ('r' in attrs):
+#        elif self.in_sheet and (name == 'row' or (has_namespace and name.endswith(':row'))) and ('r' in attrs) and not ('hidden' in attrs and attrs['hidden'] == '1'):
             self.rowNum = attrs['r']
             self.in_row = True
             self.colIndex = 0
